internal/cli/release_list.go

```diff
@@ -0,0 +1,391 @@
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/waypoint/internal/version"
+
+	"github.com/dustin/go-humanize"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/posener/complete"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	clientpkg "github.com/hashicorp/waypoint/internal/client"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	serversort "github.com/hashicorp/waypoint/internal/server/sort"
+)
+
+type ReleaseListCommand struct {
+	*baseCommand
+
+	flagWorkspaceAll bool
+	flagVerbose      bool
+	flagUrl          bool
+	flagJson         bool
+	flagId           idFormat
+	filterFlags      filterFlags
+}
+
+func (c *ReleaseListCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithSingleApp(),
+	); err != nil {
+		return 1
+	}
+
+	// Get our API client
+	client := c.project.Client()
+
+	err := c.DoApp(c.Ctx, func(ctx context.Context, app *clientpkg.App) error {
+		var wsRef *pb.Ref_Workspace
+		if !c.flagWorkspaceAll {
+			wsRef = c.project.WorkspaceRef()
+		}
+
+		// Get the latest release
+		release, err := client.GetLatestRelease(ctx, &pb.GetLatestReleaseRequest{
+			Application: app.Ref(),
+			Workspace:   c.project.WorkspaceRef(),
+		})
+		if status.Code(err) == codes.NotFound {
+			err = nil
+			release = nil
+		}
+		if err != nil {
+			app.UI.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return ErrSentinel
+		}
+
+		// List builds
+		resp, err := client.UI_ListReleases(c.Ctx, &pb.UI_ListReleasesRequest{
+			Application: app.Ref(),
+			Workspace:   wsRef,
+		})
+		if err != nil {
+			if s, ok := status.FromError(err); ok {
+				if s.Code() == codes.Unimplemented {
+					var serverVersion string
+					serverVersionResp := c.project.ServerVersion()
+					if serverVersionResp != nil {
+						serverVersion = serverVersionResp.Version
+					}
+
+					var clientVersion string
+					clientVersionResp := version.GetVersion()
+					if clientVersionResp != nil {
+						clientVersion = clientVersionResp.Version
+					}
+
+					c.project.UI.Output(
+						fmt.Sprintf("This CLI version %q is incompatible with the current server %q - missing UI_ListReleases method. Upgrade your server to v0.5.0 or higher or downgrade your CLI to v0.4 or older.", clientVersion, serverVersion),
+						terminal.WithErrorStyle(),
+					)
+					return ErrSentinel
+				}
+			}
+			c.project.UI.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return ErrSentinel
+		}
+		sort.Sort(serversort.ReleaseBundleCompleteDesc(resp.Releases))
+
+		if c.flagJson {
+			return c.displayJson(resp.Releases)
+		}
+
+		headers := []string{
+			"", "ID", "Deployment ID", "Platform", "Details", "Started", "Completed", "Health", "URL",
```

Comment:
```suggestion
			"", "ID", "Deployment ID", "Platform", "Details", "Started", "Completed", "Health",
```

The URL header gets added later!